unblob/extractor.py

```diff
@@ -46,6 +47,45 @@ def fix_permissions(outdir: Path):
             path.chmod(0o664)
 
 
+def is_safe_path(basedir: Path, path: Path) -> bool:
+    try:
+        basedir.joinpath(path).resolve().relative_to(basedir.resolve())
+    except ValueError:
+        return False
+    return True
+
+
+def fix_symlink(path: Path, outdir: Path) -> Path:
+    """Fix symlinks by rewriting absolute symlinks to make them point within
+    the extraction directory (outdir), if it's not a relative symlink it is
+    either removed it it attempts to traverse outside of the extraction directory
+    or rewritten to be fully portable (no mention of the extraction directory
+    in the link value)."""
+    target = Path(os.readlink(path))
+
+    if target.is_absolute():
+        target = Path(target.as_posix().lstrip("/"))
```

Comment:
I honestly don't understand this. Why can't we use `path.resolve()` whether it is absolute or not since later we check that if it is relative to the outdir anyways?

Reply:
So the idea is that calling `resolve()` will resolve the full path, which will always be absolute due to pathlib inner working.

Quick illustration with two symlinks extracted in `outdir = '/tmp/out'`

1. link points to `/etc/passwd`, resolving the path will give `/etc/passwd` (absolute)
2. link points to `dir_a/file_1`, resolving the path will give `/tmp/out/dir_a/file_1` (absolute)

That's why I call `os.readlink()` rather than `path.resolve()`. It's the only decent way I found to fix symlinks by rewriting **actual** absolute symlinks to make them point within the extraction directory.

Makes sense ? I may be missing an obvious way to do that in a cleaner way :)

Reply:
Yes, makes sense! I can't think of a cleaner way. I was thinking to play around with the `path.relative_to` function, but I think it would not help with symlinks pointing to non-existing targets. So this is fine by me.